helix-term/src/args.rs

```diff
@@ -41,15 +41,31 @@ impl Args {
                         }
                     }
                 }
-                arg => args.files.push(PathBuf::from(arg)),
+                arg => args.files.push(parse_file(arg)),
             }
         }
 
         // push the remaining args, if any to the files
-        for filename in iter {
-            args.files.push(PathBuf::from(filename));
+        for arg in iter {
+            args.files.push(parse_file(arg));
         }
 
         Ok(args)
     }
 }
+
+/// Parse arg into [`PathBuf`] and position.
+pub(crate) fn parse_file(s: &str) -> (PathBuf, usize, usize) {
+    split_path_pos(s).unwrap_or_else(|| (PathBuf::from(s), 0, 0))
+}
+
+/// Split file.rs:10:2 into [`PathBuf`] and position.
+///
+/// Does not validate if file.rs is a file or directory.
+fn split_path_pos(s: &str) -> Option<(PathBuf, usize, usize)> {
```

Comment:
Would it be desirable / acceptable to parse `filename:10` as `filename:10:0`? Sometimes, I know what line I want to go to (e.g. by grepping through some directory), but I don't care about the column. Sure, it's just three extra keystrokes, but I, personally, think it would be nice.

Reply:
`filename:10:0` won't work since it should be `1`, it is one indexed, not zero indexed.

Also, I don't think I know any use case that provide the output in the format of `filename:row`, even with grep it outputs it separately so I don't know if it is useful for it to be here.

Unless there is like something that outputs `filename:row` directly then I think it's worth implementing, otherwise here still needs some effort to type out the filename + pos.

Reply:
Fixed